megatron/core/models/gpt/gpt_layer_specs.py

```diff
@@ -110,9 +110,10 @@ def get_gpt_layer_with_transformer_engine_spec(
                         core_attention=TEDotProductAttention,
                         linear_proj=TERowParallelLinear,
                         # TENorm significantly harms convergence when used
-                        # for QKLayerNorm; we instead use the Apex implementation.
-                        q_layernorm=FusedLayerNorm if qk_layernorm else IdentityOp,
-                        k_layernorm=FusedLayerNorm if qk_layernorm else IdentityOp,
+                        # for QKLayerNorm; we instead use the Apex implementation (or pytorch
+                    # one if Apex is not installed).
+                        q_layernorm=LNImpl if qk_layernorm else IdentityOp,
```

Comment:
I saw that at a few other places in the code ([here](https://github.com/NVIDIA/Megatron-LM/blob/f08ab8e5ef3846505bef21853b885056645455c9/megatron/core/models/gpt/gpt_layer_specs.py#L90-L91), [here](https://github.com/NVIDIA/Megatron-LM/blob/f08ab8e5ef3846505bef21853b885056645455c9/megatron/core/inference/modelopt_support/gpt/model_specs.py#L41-L42) and [here](https://github.com/NVIDIA/Megatron-LM/blob/f08ab8e5ef3846505bef21853b885056645455c9/megatron/core/models/multimodal/llava_spec.py#L51-L52) ), `TENorm` is still used for qk-normalization (even if according to the comment above and this [commit](https://github.com/NVIDIA/Megatron-LM/commit/928aa37e7e396d8bcdf997b234dc4537c616b7a6), using `TENorm`  for qk-layernorm is unstable). 
Let me know if I should also modify these other places :+1:

Reply:
in my case, i exactly to same patch for my own megatron fork. 
so changes in this PR looks good to me, but i think we should clarify why it's happening? 
like you said, someone still use tenorm for qknorm but model converges but in my case, model doesnt converge.

Reply:
@SeunghyunSEO thanks! 
Regarding the clarification on why this is happening, do you mean that we should check why the TE implementation is diverging ? (I didn't try it myself, I just assumed it does based on your PR and also based on the comment in this [commit](https://github.com/NVIDIA/Megatron-LM/commit/928aa37e7e396d8bcdf997b234dc4537c616b7a6))

Reply:
> @SeunghyunSEO thanks! 
> 
> Regarding the clarification on why this is happening, do you mean that we should check why the TE implementation is diverging ? (I didn't try it myself, I just assumed it does based on your PR and also based on the comment in this [commit](https://github.com/NVIDIA/Megatron-LM/commit/928aa37e7e396d8bcdf997b234dc4537c616b7a6)) 

i mean if when additional feature is added, at least we should know whether it is necessary or not.
any megatron or TE maintainers know why TEnorm for qk norm diverge sometimes??? i cc sir deepak because he is the only one i communicate with! @deepakn94

Reply:
I see, makes sense I agree :+1:  Thanks for tagging @deepakn94 :pray: 
Also I think Mike Chrzanowski and Shanmugam Ramasamy can be tagged if Nvidia folks know their contact ? (because I couldn't find their github handle) 
Because they are the one who created this [commit](https://github.com/NVIDIA/Megatron-LM/commit/928aa37e7e396d8bcdf997b234dc4537c616b7a6) which prevented the use of TENorm, and also Mike Chrzanowski wrote a [paper](https://arxiv.org/abs/2410.16682) using qk-layernorm :+1: